### PR TITLE
Disable magic link login if SMTP host is not configured

### PR DESCRIPTION
**Pull request by moraj-turing**
This is a simple one! Added a basic condition in the django template to check if `EMAIL_HOST` is set to render the magic link form.

### CODE REVIEW

File: hc/accounts/views.py [hunk 1]
```diff
@@ -208,6 +208,7 @@ def login(request: HttpRequest) -> HttpResponse:
         "registration_open": settings.REGISTRATION_OPEN,
         "support_email": settings.SUPPORT_EMAIL,
         "account_closed": "account-closed" in request.GET,
+        "magic_link_enabled": settings.EMAIL_HOST not in (None, ""),
```
Comment (cuu508) [inline hc/accounts/views.py:211]: My naming preference, used in various places in this project, is `use_<somefeature>`. In this case, I suggest `"use_magic_form": ...`.

`settings.EMAIL_HOST not in (None, "")` will miss another falsy value, `False`. I think we should test for a falsy value like this instead: `bool(settings.EMAIL_HOST)`.

File: templates/accounts/login.html [hunk 2]
```diff
@@ -29,7 +29,8 @@ <h1>Sign In to {{ site_name }}</h1>
         </div>
         {% endif %}
 
-        <div class="row">
+        <div class="row" style="display: flex; justify-content: center;">
```
Comment (cuu508) [inline templates/accounts/login.html:32]: Please move the inline CSS to a CSS file.

**Review by cuu508 — CHANGES_REQUESTED** [review]
Thanks for the PR! There are a couple small things I'd like to change.

File: templates/accounts/login.html (hunk 2 above)
Comment (moraj-turing) [reply templates/accounts/login.html:32]: So given its just two rules, I tried implementing this with bootstrap classes but bootstrap has flexbox support by default until v4+. I tried a drop-in replacement but that didn't help either.

**Comment by moraj-turing** [comment]
Changes done!

**Comment by cuu508** [comment]
Thanks for making the changes. I just tested them locally and noticed another thing that needs to be addressed: if the viewport is narrow (say, phone screen), the two login forms do not stack vertically any more (I think due to the addition of the `display: flex`).

I think the way to work around this is to remove the CSS changes, and center the form using [bootstrap's offset classes](https://getbootstrap.com/docs/3.3/css/#grid-offsetting). If EMAIL_HOST is not set then the magic login form will be absent. To make sure the email/password form is centered, we can do something like:

```
<div class="col-sm-6{% if not use_magic_form%} col-sm-offset-3{% endif %}">
```

**Comment by cuu508** [comment]
Fixed in https://github.com/healthchecks/healthchecks/commit/1bdfbac775f93783f7500503ad9662faf0dfe7ca

**Pull request closed without merging**